tests/config-array.test.js

```diff
@@ -1580,6 +1816,56 @@ describe('ConfigArray', () => {
 				expect(configs.isDirectoryIgnored(path.join(basePath, 'node_modules/package/'))).to.be.true;
 			});
 
+			it('should return false when a directory is later negated with **', () => {
+				configs = new ConfigArray([
+					{
+						files: ['**/*.js']
+					},
+					{
+						ignores: [
+							'**/node_modules/**'
+						]
+					},
+					{
+						ignores: [
+							'!node_modules/**'
+						]
+					}
+				], {
+					basePath
+				});
+
+				configs.normalizeSync();
+
+				expect(configs.isDirectoryIgnored(path.join(basePath, 'node_modules'))).to.be.false;
+				expect(configs.isDirectoryIgnored(path.join(basePath, 'node_modules/'))).to.be.false;
+			});
+
+			it('should return true when a directory content is later negated with *', () => {
+				configs = new ConfigArray([
+					{
+						files: ['**/*.js']
+					},
+					{
+						ignores: [
+							'**/node_modules/**'
+						]
+					},
+					{
+						ignores: [
+							'!node_modules/*'
+						]
+					}
+				], {
+					basePath
+				});
+
+				configs.normalizeSync();
+
+				expect(configs.isDirectoryIgnored(path.join(basePath, 'node_modules'))).to.be.true;
+				expect(configs.isDirectoryIgnored(path.join(basePath, 'node_modules/'))).to.be.true;
+			});
```

Comment:
This one was failing before this change (the directory was considered not ignored).